source/Engine.cpp

```diff
@@ -311,6 +309,8 @@ void Engine::Place()
 			}
 			else if(hasOwnPlanet)
 				pos = object->Position() + angle.Unit() * Random::Real() * object->Radius();
+			else
```

Comment:
I feel that this else needs some more conditions, since we only want this new behaviour for ships that are owned by the player or escorting the player and that cannot land with the player.